core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java

```diff
@@ -341,4 +351,268 @@ public static TableInfo deserialize(byte[] bytes) throws IOException {
     return parentRelationIdentifiers;
   }
 
+  public static TableInfoBuilder builder() {
+    return new TableInfo.TableInfoBuilder();
+  }
+
+  public static class TableInfoBuilder {
```

Comment:
Can you move to new file, looks more code in inner class

Reply:
I prefer this style because they are used together (TableInfo and TableInfoBuilder), some style is used for TableSchema also